### PR TITLE
VCS Addons: histogram & polar

### PR DESCRIPTION
**Pull request by chaosphere2112**
Fixes up the existing implementation of the vcsaddons histogram and adds a new plot type (Polar) to vcsaddons. The histogram came out of needs of the GUI, and the polar plot came from some visualization tasks I had been assigned. New baselines are at UV-CDAT/uvcdat-testdata#138, here's the coverage summary for these two modules:

screen shot 2016-05-31 at 12 00 26 pm
screen shot 2016-05-31 at 12 00 30 pm

### CODE REVIEW

**Comment by chaosphere2112** [comment]
@aashish24 @doutriaux1 Please review, I want this in 2.6.

**Comment by doutriaux1** [comment]
@chaosphere2112 is merging @danlipsa revert commit branch going to affect this?

**Comment by doutriaux1** [comment]
@chaosphere2112 please merge master in

**Comment by chaosphere2112** [comment]
@doutriaux1 Merged, running tests now.

**Comment by chaosphere2112** [comment]
Looks good. Also, the commit from @sampsonbryce only made it so that calling `x.plot(s, vcsaddon)` worked, where in these tests I do `vcsaddon.plot(s, x=x)`

File: testing/vcsaddons/test_vcs_addons_polar.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import sys,os
+src = sys.argv[1]
+import testing.regression as regression
+import vcs
+import vcsaddons, numpy
+
+x=regression.init()
```
Comment (aashish24) [inline testing/vcsaddons/test_vcs_addons_polar.py]: nit pick: space before and after `=`

**Comment by aashish24** [comment]
@chaosphere2112 does this image looks right to you:

https://raw.githubusercontent.com/UV-CDAT/uvcdat-testdata/ec8b357e7fe59ab96712279f406ac2c59b9a8b43/baselines/vcsaddons/test_vcs_addons_histogram_inherit.png

The pattern seems bit different

File: Packages/vcsaddons/Lib/polar.py [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+                t_labels = None
+
+            for t in tick_thetas:
+                angle = self.theta_from_value(t)
+                x0 = center[0] + (xmul * radius * numpy.cos(angle))
+                x1 = center[0]
+                y0 = center[1] + (ymul * radius * numpy.sin(angle))
+                y1 = center[1]
+                if t_labels is not None:
+                    label = self.create_text(template.xlabel1.texttable, self.text_orientation_for_angle(angle, source=template.xlabel1.textorientation))
```
Comment (aashish24) [inline Packages/vcsaddons/Lib/polar.py]: Can we wrap these long lines please (that are > 120 column)

**Comment by aashish24** [comment]
@chaosphere2112 I did some basic check, couldn't verify all the math (specially the edge cases) but this seems like a nice addition. I just some very minor suggestions, please have  look.

**Comment by chaosphere2112** [comment]
@aashish24 Updated style issues. The baseline you linked looks fine; if you're referring to the line stippling, the issue is just that the line stipples are overlapping (each bar has its own outline), so they look solid.

**Comment by aashish24** [comment]
Okay, thanks, LGTM now :+1: Do I need to build and test?

**Comment by chaosphere2112** [comment]
@aashish24 I ran the tests and they passed for me, but, probably :wink:

**Comment by aashish24** [comment]
Okay running it now.

**Comment by aashish24** [comment]
@chaosphere2112 my build is done but couldn't finish the testing, will do it tonight.

**Comment by aashish24** [comment]
@chaosphere2112 for some reason I got more than expected test failures 😢. Let me look into it and I will ping you asap.

**Comment by aashish24** [comment]
@chaosphere2112 the baselines were not uptodate in the branch and thats why tests were failing. I am running the suite again.

**Comment by aashish24** [comment]
Also, I have merged the master baselines to this branch and running the tests again.

**Comment by aashish24** [comment]
these tests are failing but I am going to ignore dv3d and diag tests as they have been failing on master of this branch too. 

 40 - flake8_vcs (Failed)
	 60 - test_vcs_read_old_scr (Timeout)
	 62 - test_vcs_read_old_scr_2 (Timeout)
	 65 - test_vcs_no_extra_elements (Failed)
	542 - dv3d_vector_test (Failed)
	543 - dv3d_slider_test (Failed)
	544 - Hovmoller_volume_test (Failed)
	545 - dv3d_volume_test (Failed)
	546 - dv3d_constituents_test (Failed)
	547 - dv3d_surface_test (Failed)
	548 - dv3d_hovmoller_test (Failed)
	594 - CDMS_Test_multiple_formats (Failed)
	606 - diags_test_02 (Failed)
	607 - diags_test_03 (Failed)
	608 - diags_test_04 (Failed)
	609 - diags_test_41 (Failed)
	610 - diags_test_05 (Failed)
	611 - diags_test_06 (Failed)
	612 - diags_test_07 (Failed)
	613 - diags_test_08 (Failed)
	614 - diags_test_09 (Failed)
	615 - diags_test_10 (Failed)
	616 - diags_test_11 (Failed)
	617 - diags_test_12 (Failed)
	618 - diags_test_13 (Failed)
	619 - diags_test_15 (Failed)

Looking into other tests failure making sure that they are not related.

**Comment by aashish24** [comment]
@chaosphere2112 looks like polared_one broke one test: 

57: Test command: /home/chaudhary/tools/uvcdat/build_debug/install/bin/runtest "/home/chaudhary/tools/uvcdat/build_debug/install/bin/python" "/home/chaudhary/tools/uvcdat/src.git/testing/vcs/test_vcs_no_extra_elements.py"
57: Environment variables: 
57:  UVCDAT_ANONYMOUS_LOG=no
57: Test timeout computed to be: 1500
57: PATH env variable reset
57: LD_LIBRARY_PATH env variable reset
57: DYLD_LIBRARY_PATH env variable reset
57: PYTHONPATH env variable reset
57: OPAL_PREFIX env variable reset
57: LIBOVERLAY_SCROLLBAR env variable reset
57: Successfully updated your environment to use UVCDAT
57: (changes are valid for this session/terminal only)
57: Version: v2.5.0-407-ga90268e
57: Location: /home/chaudhary/tools/uvcdat/build_debug/install
57: Reset these changes by running: source /home/chaudhary/tools/uvcdat/build_debug/install/bin/reset_runtime.sh
57: template ['polar_oned']
57: textorientation ['__textorientation_21042658900488']
57: line ['__line_320635056441535', '__line_957320724349839']
57: Traceback (most recent call last):
57:   File "/home/chaudhary/tools/uvcdat/src.git/testing/vcs/test_vcs_no_extra_elements.py", line 35, in <module>
57:     raise Exception("New elements added when it shouldn't")
57: Exception: New elements added when it shouldn't
1/1 Test #57: test_vcs_no_extra_elements .......***Failed    1.64 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   1.68 sec

**Comment by chaosphere2112** [comment]
@aashish24 Ah, I could see that. I fixed the flake8 issue, I'll address the polar_oned thing.

**Comment by chaosphere2112** [comment]
Take another shot at it, @aashish24

**Comment by aashish24** [comment]
thanks @chaosphere2112 running the suite from scratch again with your latest changes.

**Comment by aashish24** [comment]
@chaosphere2112, apologies, I am pushing a fix since lot many tests were failing on my system. Also, I will open a PR against uvcdat branch.

**Pull request closed without merging**